src/include/rados/librados.h

```diff
@@ -698,6 +707,37 @@ CEPH_RADOS_API int rados_wait_for_latest_osdmap(rados_t cluster);
  */
 CEPH_RADOS_API int rados_pool_list(rados_t cluster, char *buf, size_t len);
 
+/**
+ * List pools
+ *
+ * Gets a list of pool id and names as NULL-terminated strings.
+ *
+ * If len is too short to fit all the pool name entries we need, we will fill
+ * as much as we can.
+ *
+ * Buf may be null to determine the buffer size needed to list all pools.
+ *
+ * @param cluster cluster handle
+ * @param pools output buffer
+ * @param len output buffer length
+ * @returns length of the buffer we would need to list all pools
+ */
+CEPH_RADOS_API int rados_pool_list2(rados_t cluster, rados_pool_spec_t *pools, size_t len);
```

Comment:
i am curious. we can always use the combination of `rados_pool_list()` and `rados_pool_lookup()` do get the job, as i assume this kind of operation should not be the bottleneck of your application. could you elaborate your use case a little bit?

Reply:
thanks @tchaikov .In this way, it is more convenient to query pool id and name. At the same time, 'ceph osd lspools' can also show pool id and name.